components/skus/renderer/sdk_render_frame_observer.cc

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/components/skus/renderer/sdk_render_frame_observer.h"
+
+#include <string>
+#include <vector>
+
+#include "base/feature_list.h"
+#include "base/no_destructor.h"
+#include "brave/components/skus/common/features.h"
+#include "content/public/renderer/render_frame.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+
+namespace skus {
+
+SdkRenderFrameObserver::SdkRenderFrameObserver(
+    content::RenderFrame* render_frame,
+    int32_t world_id)
+    : RenderFrameObserver(render_frame), world_id_(world_id) {}
+
+SdkRenderFrameObserver::~SdkRenderFrameObserver() {}
+
+void SdkRenderFrameObserver::DidStartNavigation(
+    const GURL& url,
+    absl::optional<blink::WebNavigationType> navigation_type) {
+  url_ = url;
+}
+
+void SdkRenderFrameObserver::DidCreateScriptContext(
+    v8::Local<v8::Context> context,
+    int32_t world_id) {
+  if (!render_frame()->IsMainFrame() || world_id_ != world_id)
+    return;
+
+  // There could be empty, invalid and "about:blank" URLs,
+  // they should fallback to the main frame rules
+  if (url_.is_empty() || !url_.is_valid() || url_.spec() == "about:blank")
+    url_ = url::Origin(render_frame()->GetWebFrame()->GetSecurityOrigin())
+               .GetURL();
+
+  if (!IsSkusSdkAllowed())
+    return;
+
+  if (!native_javascript_handle_) {
+    native_javascript_handle_.reset(new SdkPageController(render_frame()));
```

Comment:
`native_javascript_handle_ = std::make_unique<SdkPageController>(render_frame());`

Reply:
Nice find! Fixed in https://github.com/brave/brave-core/pull/10847/commits/db48260dd1e9b819e4d991e99f8dcb3c31ca364c